gateway/gateway.go

```diff
@@ -840,7 +839,6 @@ func (gateway *HandleT) Setup(application app.Interface, backendConfig backendco
 
 	if enableDedup {
 		gateway.openBadger(clearDB)
-		defer gateway.badgerDB.Close()
```

Comment:
Earlier, we had a forever blocking call at the end. So that would trigger the badgerDB close.

If we remove this altogether, then we might leak connections to badgerDB. It then depends on how badger closed the pending connections.

Reply:
@fxenik @psrikanth88  Fotis refactored these during the unit tests and might have removed the blocking channel wait. We need to see what is the right fix for this.

One option is to call gateway.Setup() in a goroutine and have a blocking call at the end inside it.

Reply:
Earlier, we had a `log.Fatal(http.ListenAndServe(..` which dies with an `os.Exit()`
Even that wouldn't have honored the `defer`.